template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/adls/AdlsGen2ConfigProvider.java

```diff
@@ -0,0 +1,28 @@
+package com.sequenceiq.cloudbreak.cmtemplate.configproviders.adls;
+
+import static com.sequenceiq.cloudbreak.cmtemplate.CMRepositoryVersionUtil.isVersionNewerOrEqualThanLimited;
+import static com.sequenceiq.common.model.FileSystemType.ADLS_GEN_2;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cmtemplate.CMRepositoryVersionUtil;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils;
+import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
+
+@Component
+public class AdlsGen2ConfigProvider {
+
+    private static final String AZURE_READAHEAD_ENABLED = "fs.azure.enable.readahead";
+
+    private static final String AZURE_READAHEAD_ENABLED_VALUE = "false";
+
+    public void populateServiceConfigs(TemplatePreparationObject source, StringBuilder hdfsCoreSiteSafetyValveValue, String stackVersion) {
+        if (adlsGen2FileSystemConfigured(source) && isVersionNewerOrEqualThanLimited(stackVersion, CMRepositoryVersionUtil.CLOUDERA_STACK_VERSION_7_2_12)) {
+            hdfsCoreSiteSafetyValveValue.append(ConfigUtils.getSafetyValveProperty(AZURE_READAHEAD_ENABLED, AZURE_READAHEAD_ENABLED_VALUE));
```

Comment:
We are not allowing safety valves, is there any way to use a proper config?

Reply:
The cloud storage connectors team mentioned that this config has to be added to core-site.xml. I'm not sure if regular configs are propagated to the `Advanced Configuration Snippet` in CM for core-site, hence I added it to this safety valve.

Reply:
>I'm not sure if regular configs
They should, please reach out to CM team, and if for some reason there is no way to pass this through the CM API, then make sure that there is a ticket raised against CM.

Reply:
@akanto this config is not available in CM. CM has configs for only the commonly updated ones. Hadoop stack(across all services) has a huge number of configs, there is no way that we can expose all of them through CM

We can file a CM jira to add this specific config, but that will require another runtime release. Lets go ahead with this safety valve config please

Reply:
Many thanks, please file a ticket, until that no other option than safety valve,

Reply:
Filed: https://jira.cloudera.com/browse/OPSAPS-65427